backend/inspirehep/records/api/literature.py

```diff
@@ -421,7 +423,16 @@ def add_file(
                 return result
             file_data = download_file_from_url(url)
             new_key = hash_data(file_data)
+
             mimetype = magic.from_buffer(file_data, mime=True)
+            mimetype_guess, _ = guess_type(url)
```

Comment:
I don't think this will be very effective for the main use-case (avoiding invalid PDFs). If the `url` doesn't end in `.pdf` (which I expect happens pretty often, e.g. on INSPIRE), this won't do anything. Maybe we should additionally maintain a black/whitelist?

Reply:
it's not at all, but after the discussion with Thorsten there is no way to have a blacklist, we should accept any kind of files actually. Another solution will be to verify the response `content-type` but still if it supposes to be a `pdf` and the url is `text/html` and we check the content of the url and it's `html` then it's exactly the same issue.

Reply:
what I meant to convey is that you can't trust the content-type, extension, mime-type or any number of things. you have to verify the format of the file once you have a local copy.
for PDF, PS, png, jpg, gif there are tools to not only check that the file has the proper "magic sequence" in the first few dozen bytes of the file (python-magic),
but that the entire file is syntactically correct. think pylint or xmllint for other file types.
Specifically for PDF, you can run `gs -q -DNODISPLAY <file> -c quit` or PyPDF2 open or some such (e.g. mutool). Similarly the most commonly used figure formats have a variety of tools, for example PIL's `im.verify()`

Reply:
never trust external sources, always verify